### PR TITLE
64KB block and chip erase for Nordic nrf52 QSPI

### PR DESCRIPTION
**Pull request by mikevoyt**
## Description
Without a hff_erase implementation for HAL flash functions, the HAL flash layer will default to sector erase for all
erase functions, regardless of length.

This PR implements hff_erase, and will issue a chip erase or 64KB block erase where possible based on the address and length of area to be erased.

## How was this tested?

On the nRF52840 DK (PCA10056) using the provided MX25R6435F QSPI flash, I measured a 4K sector erase of about 84 mS.  With this patch, a 64KB block erase averages about 1164 mS.  This saves about 176 mS per 64K erase.

### CODE REVIEW

File: hw/mcu/nordic/nrf52xxx/src/hal_qspi.c [hunk 1]
```diff
@@ -226,16 +246,53 @@ nrf52k_qspi_erase_sector(const struct hal_flash *dev,
...
+    address &= ~0xFFFU;
+    end = address + size;
+
+    if (end == MYNEWT_VAL(QSPI_FLASH_SECTOR_COUNT) *
+                   MYNEWT_VAL(QSPI_FLASH_SECTOR_SIZE)) {
+        return erase_block(0, NRF_QSPI_ERASE_LEN_ALL);
+    }
+
+    while (size) {
+#if MYNEWT_VAL(SPIFLASH_BLOCK_ERASE_64BK)
```
Comment (andrzej-kaczmarek) [inline hw/mcu/nordic/nrf52xxx/src/hal_qspi.c:272]: this is syscfg from different package, do not use it here. either make it always enabled or alternatively add new syscfg to MCU package

Comment (mikevoyt) [reply hw/mcu/nordic/nrf52xxx/src/hal_qspi.c]: @andrzej-kaczmarek I've removed the `SPIFLASH_BLOCK_ERASE_64BK` conditional.

File: hw/mcu/nordic/nrf52xxx/src/hal_qspi.c [hunk 2]
```diff
@@ -218,30 +223,77 @@ nrf52k_qspi_write(const struct hal_flash *dev, uint32_t address,
...
+    return 0;
+}
+
+static int
+nrf52k_qspi_erase(const struct hal_flash *dev, uint32_t address,
+                  uint32_t size)
+{
+
+    uint32_t end;
+    int rc;
```
Comment (kasjer) [inline hw/mcu/nordic/nrf52xxx/src/hal_qspi.c:262]: rc is not used any more, it should be removed

**Review by kasjer — CHANGES_REQUESTED** [review]
@mikevoyt I tested this PR and and works (once unused variable is remove).
There is something that could be done though.
First commit message was clean and it was obvious what the change is.
Next few make sense only in context of PR.
I can squash them in github but then your authorship will be messaged up.
I suggest to fix rc, then squash all commits, force push to the same branch, so we can see one clean commit that adds 64KB erase, and that will go to the git history.

File: hw/mcu/nordic/nrf52xxx/src/hal_qspi.c (hunk 2 above)
Comment (mikevoyt) [reply hw/mcu/nordic/nrf52xxx/src/hal_qspi.c]: @kasjer no problem, I've removed `int rc`, and forced pushed the branch so now we have one clean commit.

**Review by kasjer — APPROVED** [review]
Works and look great, thank you @mikevoyt

**Pull request merged**